dev-server/static/scripts/vitalsource-mosaic-book-element.js

```diff
@@ -165,6 +165,20 @@ export class MosaicBookElement extends HTMLElement {
     return this.pageData[this.pageIndex];
   }
 
+  async getTOC() {
+    // In our VS test pages, the TOC entries are 1:1 with the page entries.
+    // However in a real VS book there may be multiple TOC entries for different
+    // parts of a single page, and there is also an issue where the titles
+    // may be different. See https://github.com/hypothesis/client/issues/4986.
+    const tocEntries = this.pageData.map(page => ({
+      cfi: page.cfi,
+      title: page.chapterTitle,
+      path: page.absoluteURL,
+      page: page.page,
+    }));
+    return { ok: true, data: tocEntries, status: 200 };
```

Comment:
Can you add a comment to associate this return type with the `DataResponse` type defined in the Vitalsource integration?

Reply:
I think the best way to handle this will be to actually use JSDoc + TypeScript to link this module with the corresponding interface in `vitalsource.ts`. I had a go at this but realized it would be too big of a change to add into this PR, so I'll look at that separately.